### PR TITLE
Give keys and keypairs a name

### PR DESCRIPTION
**Pull request by ericcornelissen**
- Relevant Issues: #69, #74
- Related Pull Requests: #89, #93

* * *

## What
Allow both the `IKey` class and `KeyPair` class to have names so they're easier to identify.

## Why
This allows users the identify their keys by name instead of the key itself.

## How
Adding a `private final String` for the name to both the classes and requiring them to be set in the constructor.

## Alternative implementation
Non

## Notes
n/a

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java [hunk 1]
```diff
@@ -14,12 +15,22 @@
      * @param publicKey  The publicKey of the KeyPair.
      * @param privateKey The privateKey of the KeyPair.
      */
-    public KeyPair(final IKey publicKey, final IKey privateKey) {
+    public KeyPair(final String name, final IKey publicKey, final IKey privateKey) {
+        this.name = name;
```
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java]: Add javadoc

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java [hunk 2]
```diff
@@ -56,7 +68,7 @@ public boolean equals(final Object o) {
      */
     @Override
     public int hashCode() {
-        return this.publicKey.hashCode() + this.privateKey.hashCode();
+        return this.name.hashCode() + this.publicKey.hashCode() + this.privateKey.hashCode();
```
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java]: We should consider adding HashCodeBuilder and EqualsBuilder for next week

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java (hunk 1 above)
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java]: Good catch!

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java (hunk 2 above)
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/KeyPair.java]: I'm mot sure this is needed as that is very class specific

**Pull request merged**